modules/mapbox/src/deck-utils.ts

```diff
@@ -50,7 +51,7 @@ export function getDeckInstance({
 
   let deckInstance: Deck;
 
-  if (!deck || deck.props.gl === gl) {
+  if (!deck || (deck.props.device instanceof WebGLDevice && deck.props.device.gl === gl)) {
```

Comment:
This will break the check. Note you are checking `deck.props` which is constructed below at line 59. Only `gl` is passed, not `device`.